### PR TITLE
Added optional arguments to `/dag/put`

### PR DESCRIPTION
**Pull request by meyer1994**
I am not sure if I did this correctly. I still do not understand some parts of the code. There are lots of backawards compatibility stuff that I do not know (yet).

### CODE REVIEW

File: ipfshttpclient/client/dag.py [hunk 1]
```diff
@@ -69,6 +70,8 @@ def put(self, data: utils.clean_file_t, **kwargs: base.CommonArgs):
 			dict
```
Comment (ntninja) [inline ipfshttpclient/client/dag.py:70]: The parameter section here should document the `format` and `input_enc` parameters

**Review by ntninja — APPROVED** [review]
Missing docs for the new parameters, otherwise LGTM!

(I'm currently a bit busy, so feel free to add the comments, otherwise I'll do when I merge this.)

File: ipfshttpclient/client/dag.py (hunk 1 above)
Comment (meyer1994) [reply ipfshttpclient/client/dag.py:74]: Done :)

**Comment by ntninja** [comment]
Thanks!

**Pull request merged**